Library/Formula/lcm.rb

```diff
@@ -0,0 +1,48 @@
+class Lcm < Formula
+  desc "libraries and tools for message passing and data marshalling"
+  homepage "https://lcm-proj.github.io/"
+  url "https://github.com/lcm-proj/lcm/releases/download/v1.3.0/lcm-1.3.0.zip"
+  sha256 "23bce19a4a6dc5df444d811f57796e14b4b0d1886cb3c219f90b72f328f2184e"
+
+  head do
+    # This url can be changed to "https://github.com/lcm-proj/lcm.git"
+    # after PR #48 on lcm is accepted.
+    url "https://github.com/bluesquall/lcm.git", :branch => "simplify-java-Makefile.am"
+
+    depends_on "xz" => :build
+    depends_on "libtool" => :build
+    depends_on "automake" => :build
+    depends_on "autoconf" => :build
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on :java => :recommended
```

Comment:
I'd lean toward `:recommended`, agreeing with bluesquall's reasoning. Weighing in that Java is heavyweight, but probably likely to be already installed among LCM's main userbase, or not objectionable, so `:optional` would inconvenience most of its users and help few (those LCM users who don't have or want Java installed).

Especially since [upstream's OS X-specific build instructions](http://lcm-proj.github.io/build_instructions.html#build_osx) flat-out say to install Java, unconditionally, even though it's just "strongly recommended" in the dependency listing. If we do `:optional`, then our default installation of `lcm` deviates from upstream's standard OS X installation.

Also because the process to fix a "wrong" default installation is different: For non-Java LCM users, default installation will fail and the error message will tell them how to fix it, and then they just have to rerun with `--without-java`. For LCM users who want all the tools, default install will succeed silently but will be missing some stuff and they'll have to do some research to figure out why and how to fix it.

But this is complicated by another factor: the build boxes. The test bots on which bottles are built have the JDK installed. So the `:optional` `:java` requirement will be recognized as being met, and the bottles are built with Java support in them. So if it's `:optional`, a user can do `brew install lcm` and install it from a bottle and they will get Java support – they can get a bottle install with Java support as long as they remember to *not* specify `--with-java`. A user who doesn't have Java installed and wants to actually install LCM without Java support built will have to know to do `brew install -s lcm`, (and `brew install --without-java` won't install an actual non-Java build without `-s`) even though Java is `:optional` so you'd think that would be the default behavior. This may not matter in practice because Java-based stuff is usually done dynamically, so maybe the parts that require Java will just get turned off or be broken if you try to run them. So in practice most people will get Java support anyway, but this relies on weird edge case behavior and our build bot setup and is confusing.

You can see this in the build logs. The testing output has the `-j` Java test, so you can see that `build_with? "java"` was in effect.

```
==> /usr/local/Cellar/lcm/1.3.0/bin/lcm-gen -j example_t.lcm
```

So, maybe `:recommended` is also useful because then the bottles will line up with the formula definition, `--without-java` will always work, and the effects of `brew install lcm` options will be easier to understand. And the behavior won't change if we change our test bot configuration.